t/03-proxy_wasm/hfuncs/metrics/003-increment_metric_misuse.t

```diff
@@ -0,0 +1,60 @@
+# vim:set ft= ts=4 sts=4 sw=4 et fdm=marker:
+
+use strict;
+use lib '.';
+use t::TestWasmX;
+
+plan_tests(7);
+run_tests();
+
+__DATA__
+
+=== TEST 1: proxy_wasm metrics - increment_metric() gauge
+--- valgrind
+--- load_nginx_modules: ngx_http_echo_module
+--- wasm_modules: hostcalls
+--- config eval
+qq{
+    location /t {
+        proxy_wasm hostcalls 'on_configure=define_metrics \
+                              on=request_headers \
+                              test=/t/metrics/increment_gauges \
+                              metrics=g1,g2';
+        echo ok;
+    }
+}
+--- error_code: 500
+--- error_log eval
+[
+    qr/\[error\] .+ \[wasm\] attempt to call increment_metric on a gauge; operation not supported/,
```

Comment:
Is there a way to make this more forgiving and log a `warn` without triggering a host trap in these "bad operations" cases? Would it make sense? It would seem to both get rid of the double logging and be more forgiving. We can have a function return `NGX_ERROR` or `NGX_ABORT`, and the caller decides whether an error warrants a host trap or not (since the caller today are host functions, but may be another C function later).
The same comment for all below "bad operation" test cases.